mail-mta/sendmail/sendmail-8.16.1-r1.ebuild

```diff
@@ -0,0 +1,203 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit systemd toolchain-funcs
+
+DESCRIPTION="Widely-used Mail Transport Agent (MTA)"
+HOMEPAGE="https://www.sendmail.org/"
+SRC_URI="ftp://ftp.sendmail.org/pub/${PN}/${PN}.${PV}.tar.gz"
+
+LICENSE="Sendmail GPL-2" # GPL-2 is here for initscript
+SLOT="0"
+#KEYWORDS="~alpha ~amd64 ~arm ~hppa ~ia64 ~mips ~ppc ~ppc64 ~s390 ~sparc ~x86"
+IUSE="ipv6 ldap mbox nis sasl sockets ssl tcpd"
+
+BDEPEND="sys-devel/m4"
+DEPEND="
+	net-mail/mailbase
+	>=sys-libs/db-3.2:=
+	ldap? ( net-nds/openldap )
+	nis? ( net-libs/libnsl:= )
+	sasl? ( >=dev-libs/cyrus-sasl-2.1.10 )
+	ssl? ( dev-libs/openssl:0= )
+	tcpd? ( sys-apps/tcp-wrappers )"
+RDEPEND="
+	${DEPEND}
+	acct-group/smmsp
+	>=acct-user/smmsp-0-r2
+	>=mail-filter/libmilter-1.0.2_p1-r1
+	!mail-mta/courier
+	!mail-mta/esmtp
+	!mail-mta/exim
+	!mail-mta/mini-qmail
+	!mail-mta/msmtp[mta]
+	!mail-mta/netqmail
+	!mail-mta/nullmailer
+	!mail-mta/opensmtpd
+	!mail-mta/postfix
+	!mail-mta/qmail-ldap
+	!>=mail-mta/ssmtp-2.64-r2[mta]
+	!net-mail/vacation"
+PDEPEND="!mbox? ( mail-filter/procmail )"
+
+src_prepare() {
+	eapply "${FILESDIR}"/${PN}-8.16.1-build-system.patch
+	eapply -p0 "${FILESDIR}"/sendmail-delivered_hdr.patch
+	eapply_user
+
+	local confCCOPTS="${CFLAGS}"
+	local confENVDEF="-DMAXDAEMONS=64 -DHAS_GETHOSTBYNAME2=1"
+	local confLDOPTS="${LDFLAGS}"
+	local confLIBS=
+	local confMAPDEF="-DMAP_REGEX"
+	local conf_sendmail_LIBS=
+
+	if use ldap; then
+		confMAPDEF+=" -DLDAPMAP"
+		confLIBS+=" -lldap -llber"
+	fi
+
+	if use sasl; then
+		confLIBS+=" -lsasl2"
+		confENVDEF+=" -DSASL=2"
+		confCCOPTS+=" -I/usr/include/sasl"
```

Comment:
${ESYSROOT} given it's passed directly with -I

Reply:
Good point, didn't want to touch too many things but these are obvious.

Reply:
Then again, this is where this format gets nasty, e.g. this won't deal with spaces. Even if used an array it's latter seded into the .m4

Reply:
Turns out this one is not needed, sendmail actually uses `include <sasl/...>` so I'll swap these for a tc-pkg --cflags libsasl2 along others I can.